nixos/modules/tasks/network-interfaces-systemd.nix

```diff
@@ -53,20 +59,27 @@ in
     systemd.network =
       let
         domains = cfg.search ++ (optional (cfg.domain != null) cfg.domain);
-        genericNetwork = override:
-          let gateway = optional (cfg.defaultGateway != null) cfg.defaultGateway.address
+        defaultGateways = optional (cfg.defaultGateway != null) cfg.defaultGateway.address
             ++ optional (cfg.defaultGateway6 != null) cfg.defaultGateway6.address;
-          in {
+        genericNetwork = override:
+          {
             DHCP = override (dhcpStr cfg.useDHCP);
-          } // optionalAttrs (gateway != [ ]) {
-            gateway = override gateway;
+          } // optionalAttrs (defaultGateways != [ ]) {
+            gateway = override defaultGateways;
           } // optionalAttrs (domains != [ ]) {
             domains = override domains;
+          } // optionalAttrs (! cfg.enableIPv6) {
```

Comment:
This part is still causing problems when using vlans (because the property is merged twice, see other comments)